src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,186 @@
+		{
+			var diag = Diagnostic.Create(TextColorToDiagnostic.MauiReferenceIsMissing, Location.None);
+			context.ReportDiagnostic(diag);
+			return;
+		}
+
+		var textStyleClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace)>();
+
+		// Collect Microsoft.Maui.Controls that Implement ITextStyle
+		var mauiTextStyleImplementors = mauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols().Where(x => x.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& x.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default));
+
+		foreach (var namedTypeSymbol in mauiTextStyleImplementors)
+		{
+			textStyleClassList.Add((namedTypeSymbol.Name, "public", namedTypeSymbol.ContainingNamespace.ToDisplayString()));
+		}
+
+		// Collect All Classes in User Library that Implement ITextStyle
+		foreach (var classDeclarationSyntax in userGeneratedClassesProvider)
+		{
+			var declarationSymbol = compilation.GetSymbol<INamedTypeSymbol>(classDeclarationSyntax);
+			if (declarationSymbol is null)
+			{
+				var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidClassDeclarationSyntax, Location.None, classDeclarationSyntax.Identifier.Text);
+				context.ReportDiagnostic(diag);
+				continue;
+			}
+
+			if (declarationSymbol.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& declarationSymbol.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default))
+			{
+				var nameSpace = declarationSymbol.ContainingNamespace.ToDisplayString();
+				if (nameSpace == "<global namespace>")
+				{
+					var diag = Diagnostic.Create(TextColorToDiagnostic.GlobalNamespace, Location.None, declarationSymbol.Name);
+					context.ReportDiagnostic(diag);
+					continue;
+				}
+
+				var accessModifier = GetClassAccessModifier(declarationSymbol);
+
+				if (accessModifier == string.Empty)
+				{
+					var diag = Diagnostic.Create(TextColorToDiagnostic.InvalidModifierAccess, Location.None, declarationSymbol.Name);
+					context.ReportDiagnostic(diag);
+					continue;
+				}
+
+				textStyleClassList.Add((declarationSymbol.Name, accessModifier, nameSpace));
+			}
+		}
+
+		var options = ((CSharpCompilation)compilation).SyntaxTrees[0].Options as CSharpParseOptions;
+		foreach (var textStyleClass in textStyleClassList)
+		{
+			var textColorToBuilder = @"
+// AutoGenerated Code
+// See: CommunityToolkit.Maui.SourceGenerators.TextColorToGenerator
+
+using System;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.Core.Extensions;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
+
+namespace " + textStyleClass.Namespace + @";
+
+" + textStyleClass.ClassAcessModifier + @" static partial class ColorAnimationExtensions_" + textStyleClass.ClassName + @"
+{
+	/// <summary>
+	/// Animates the TextColor of an <see cref=""Microsoft.Maui.ITextStyle""/> to the given color
+	/// </summary>
+	/// <param name=""element""></param>
```

Comment:
Missing documentation for this parameter. Not sure if the suggestion is accurate.

```suggestion
	/// <param name=""element"">The element for which the TextColor will be animated.</param>
```

Reply:
cc: @brminnick